Common/Core/RecoDecay.h

```diff
@@ -663,26 +665,33 @@ struct RecoDecay {
   /// Checks whether the reconstructed decay candidate is the expected decay.
   /// \param checkProcess  switch to accept only decay daughters by checking the production process of MC particles
   /// \param acceptIncompleteReco  switch to accept candidates with only part of the daughters reconstructed
+  /// \param acceptTrackDecay  switch to accept candidates with one or more tracks decayed during transport
   /// \param particlesMC  table with MC particles
   /// \param arrDaughters  array of candidate daughters
   /// \param PDGMother  expected mother PDG code
   /// \param arrPDGDaughters  array of expected daughter PDG codes
   /// \param acceptAntiParticles  switch to accept the antiparticle version of the expected decay
   /// \param sign  antiparticle indicator of the found mother w.r.t. PDGMother; 1 if particle, -1 if antiparticle, 0 if mother not found
   /// \param depthMax  maximum decay tree level to check; Daughters up to this level will be considered. If -1, all levels are considered.
+  /// \param nPiToMu  number of pions decayed in a muon during transport
+  /// \param nKaToPi  number of kaons decayed in a pion during transport
   /// \return index of the mother particle if the mother and daughters are correct, -1 otherwise
-  template <bool acceptFlavourOscillation = false, bool checkProcess = false, bool acceptIncompleteReco = false, std::size_t N, typename T, typename U>
+  template <bool acceptFlavourOscillation = false, bool checkProcess = false, bool acceptIncompleteReco = false, bool acceptTrackDecay = false, std::size_t N, typename T, typename U>
   static int getMatchedMCRec(const T& particlesMC,
                              const std::array<U, N>& arrDaughters,
                              int PDGMother,
                              std::array<int, N> arrPDGDaughters,
                              bool acceptAntiParticles = false,
                              int8_t* sign = nullptr,
-                             int depthMax = 1)
+                             int depthMax = 1,
+                             int8_t* nPiToMu = nullptr,
+                             int8_t* nKaToPi = nullptr)
   {
     // Printf("MC Rec: Expected mother PDG: %d", PDGMother);
     int8_t coefFlavourOscillation = 1;     // 1 if no B0(s) flavour oscillation occured, -1 else
     int8_t sgn = 0;                        // 1 if the expected mother is particle, -1 if antiparticle (w.r.t. PDGMother)
+    int8_t piToMu = 0;                     // number of pions decayed to a muon >=1 if any are found
+    int8_t kaToPi = 0;                     // number of kaons decayed to a pion >=1 if any are found
```

Comment:
```suggestion
    if constexpr (acceptTrackDecay) {
      int8_t nPiToMuLocal = 0; // number of pion prongs decayed to a muon
      int8_t nKaToPiLocal = 0; // number of kaon prongs decayed to a pion
    }
```

Reply:
Testing locally this change causes the compilation to break. Can I define these variables outside the if statement?